cmd/rpcdaemon/cli/config.go

```diff
@@ -449,10 +440,23 @@ func RemoteServices(ctx context.Context, cfg *httpcfg.HttpCfg, logger log.Logger
 			}
 		}
 		stateCache = kvcache.NewDummy()
-	}
-	// If DB can't be configured - used PrivateApiAddr as remote DB
-	if db == nil {
+	} else {
+		// use PrivateApiAddr as remote DB
 		db = remoteKv
+
+		if err = db.View(context.Background(), func(tx kv.Tx) error {
```

Comment:
Unfortunately we can't do it - `View` will wait for server to startup. 
I think "hard startup-dependency on another process liveness" - is the path to  'cascade outage` or 'all processes waiting each other startup'. Good practice - is to startup and "do what can even if monkey is in datacenter".

`BaseAPI` already has method `chainConfig` which does caching/waiting_for_server_startup/etc... maybe can make cache persisted. maybe you have another ideas - feel free to share.

Reply:
Thanks @AskAlexSharov , good point - I totally missed that the remoteEngine is initialised in a separate background routine.

Fixed now in a simpler way that also sticks to the current way of handling this scenario.